es_ES/design-principles.md

```diff
@@ -0,0 +1,241 @@
+
+ 3. **Si la gente está de acuerdo en que los beneficios de cambiar la API superan los costos de estabilidad**, puedes continuar con el proceso normal de revisión de código para realizar cambios. Debes dejar algún tiempo entre los pasos 2 y 3 (como mínimo 24 horas durante la semana laboral para que la gente de todas las zonas horarias haya tenido la oportunidad de verlo, pero lo ideal sería una semana más o menos).
+
+ 4. Si has conseguido un breaking change, añade un punto a la sección superior de [página Changelog en la wiki](https://github.com/flutter/flutter/wiki/Changelog),
+    describiendo tu cambio y enlazando a tu correo electrónico en [los archivos de la lista de correo](https://groups.google.com/forum/#!forum/flutter-dev).
+    Para averiguar la versión correcta para del encabezado para el changelog
+    `git fetch upstream && flutter --version`. Por ejemplo, si te dice “Flutter 0.0.23-pre.10” en la salida tu entrada en el changelog debería estar bajo el encabezado “Changes since 0.0.22”.
+
+En la medida de lo posible, incluso los “breaking” changes deben hacerse de una manera retrocompatible por ejemplo, introduciendo una nueva clase y marcando la clase antigua. `@deprecated`. Al hacerlo, incluya una descripción de cómo realizar la transición en la notificación de deprecación, por ejemplo:
+
+<!-- skip -->
+```dart
+@deprecated('FooInterface has been deprecated because ...; it is recommended that you transition to the new FooDelegate.')
+class FooInterface {
+/// ...
+}
+```
+
+Si utiliza `@deprecated`, asegúrate de recordar eliminar realmente la función unas semanas más tarde (después de la próxima versión beta), ¡no lo dejes para siempre!
+
+
+#### Responsabilidades exclusivas de Google
+
+Si trabajas para Google, tienes la responsabilidad añadida de actualizar la copia interna de Google de Flutter y arreglar cualquier sitio donde haya una llamada rota razonablemente rapido despues de hacer merge de los cambios en upstream.
+
+Programación lazy
+----------------
+
+Escribe lo que necesites y nada más, pero cuando lo escribas, hazlo bien.
+
+Evita implementar funciones que no necesites. No se puede diseñar una característica sin saber cuáles son las restricciones. La implementación de características "para completar" resulta en código sin usar que es costoso de mantener, conocer, documentar, probar, etc.
+
+Cuando implementes una característica, impleméntala de la manera correcta. Evita las soluciones alternativas. Las soluciones alternativas no hacen sino agravar el problema más adelante, pero a un coste mayor: alguien tendrá que volver a aprender el problema, averiguar cómo solucionarlo y desmontarlo (y todos los lugares que ahora lo utilizan), e implementar la característica. Es mucho mejor tardar más tiempo en arreglar un problema correctamente, que ser el que lo arregla todo rápidamente, pero de una manera que requerirá una limpieza posterior.
+
+Es posible que escuche a los miembros del equipo decir "embrace the [yak
```

Comment:
Es posible que escuches a los miembros del equipo decir “embrace the [yak shave](http://www.catb.org/jargon/html/Y/yak-shaving.html)!" Esto es un estímulo para tomar el esfuerzo más grande necesario para realizar una solución adecuada para un problema en lugar de simplemente aplicar soluciones paliativas.